sktime/clustering/base/base.py

```diff
@@ -60,81 +57,120 @@ def fit(self, X: NumpyOrDF, y: NumpyOrDF = None):
-        X: Numpy array
-            Numpy array to train the model on
-
-        y: Numpy array, default = None
-            Numpy array that is the labels for training.
-            Unlikely to be used for clustering but kept for consistency
+        X: 2D np.array with shape (n_instances, n_timepoints)
+            panel of univariate time series to train the clustering model on
 
         Returns
         -------
-        self
-            Fitted estimator
+        reference to self
         """
         raise NotImplementedError("abstract method")
 
     def _predict(self, X: NumpyArray) -> NumpyArray:
         """
-        Method that is used
+        Return cluster center index for data samples.
+
+            core logic
+
+        Parameters
+        ----------
+        X: 2D np.array with shape (n_instances, n_timepoints)
+            panel of univariate time series to cluster
+
+        Returns
+        -------
+        Numpy_Array: 1D np.array of length n_instances
+            Index of the cluster each sample belongs to
         """
         raise NotImplementedError("abstract method")
 
     def _check_params(self, X: NumpyArray):
         """
-        Method used to check the parameters passed
+        Custom input checking, should raise errors
 
         Parameters
         ----------
-        X: Numpy_Array
-            Dataset to be validate parameters against
+        X: 2D np.array with shape (n_instances, n_timepoints)
         """
         return
 
-    def fit_predict(self, X: NumpyOrDF) -> NumpyArray:
+    def fit_predict(self, X: NumpyOrDF, y=None) -> NumpyArray:
         """
-        Method that calls fit and then returns a prediction
-        for the value of X
+        clusters time series and returns cluster labels
 
         Parameters
         ----------
-        X: Numpy array or Dataframe
-            sktime data_frame or numpy array containing the values
-            to perform clustering algorithm on
+        X: 2D np.array with shape (n_instances, n_timepoints)
+           or pd.DataFrame in nested format
+            panel of univariate time series to cluster
+
+        y: ignored, exists for API consistency reasons
+
+        Returns
+        -------
+        Numpy_Array: 1D np.array of length n_instances
+            Index of the cluster each sample belongs to
         """
         self.fit(X)
         return self.predict(X)
 
+    def get_fitted_params(self):
+        """
+        Retrieves fitted parameters of cluster model
+
+        returns
+        ----------
+        param_dict: dictionary of fitted parameters
+        """
+        self.check_is_fitted()
+
+        return self._get_fitted_params()
+
+    def _get_fitted_params(self):
+        """
+        Retrieves fitted parameters of cluster model
+
+            core logic
+
+        returns
+        ----------
+        param_dict: dictionary of fitted parameters
+        """
+        raise NotImplementedError
+
 
 class BaseClusterCenterInitializer:
```

Comment:
Is this here for any reason or did you want me to write add templates for CenterInitializers, ClusterAverage, and ClusterApproximate?

Reply:
well, I thought there are two main reasons to do clustering:

* be able to make individual cluster assignments, on existing or new data
* look at model outputs of the clustering

For instance, in mixture clustering, you may want to look at the centroids and the individual cluster's parameters (mean and precision matrix).

That's why I thought `get_fitted_params` should be in the base class, but not every method needs to implement it.
We currently have it in the forecaster too, and not every forecaster implements it either.